frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyMfaCodeHandler.java

```diff
@@ -315,15 +309,58 @@ private APIGatewayProxyResponseEvent verifyCode(
                     session);
         }
 
-        processCodeSession(
-                errorResponseMaybe,
-                session,
-                input,
-                userContext,
-                subjectId,
-                codeRequest,
-                mfaCodeProcessor,
-                maybeRpPairwiseId);
+        var errorResponseMaybe = mfaCodeProcessor.validateCode();
+
+        if (errorResponseMaybe.isPresent()) {
+            var errorResponse = errorResponseMaybe.get();
```

Comment:
I wonder here about introducing a "process error response" method

Reply:
I had a go at that, but I found that with ERROR_1041 returning an APIGatewayProxyResponseEvent while the other errors only cause interactions with services it looks clunkier and harder to read. I might be missing something though so would be happy to hear if you have a better idea on how to do it

Reply:
Fair enough, I think my inclination would be to try a method that returns an optional gateway response, but I haven't seen how clunky that is :-) Happy to leave as is